external/storm-kafka/src/jvm/storm/kafka/KafkaStateStore.java

```diff
@@ -0,0 +1,220 @@
+package storm.kafka;
+
+import com.google.common.collect.Maps;
+import kafka.api.ConsumerMetadataRequest;
+import kafka.common.ErrorMapping;
+import kafka.common.OffsetAndMetadata;
+import kafka.common.OffsetMetadataAndError;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.*;
+import kafka.network.BlockingChannel;
+import org.json.simple.JSONValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class KafkaStateStore implements StateStore {
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaStateStore.class);
+
+    private SpoutConfig _spoutConfig;
+
+    private int _correlationId = 0;
+    // https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java
+    private volatile BlockingChannel _offsetManager;
+
+    public KafkaStateStore(Map stormConf, SpoutConfig spoutConfig) {
```

Comment:
<tt>stormConf</tt> isn't used so remove.  By using <tt>spoutConfig</tt>, you are bleeding the spout state into the store.  I understand it is for convenience, but using explicit would be better, so perhaps you can create a <tt>KafkaStoreConfig</tt>.